### PR TITLE
Updated eiface code.

### PR DESCRIPTION
**Pull request by jvelilla**
Created a library eiface.
Refactor directory structure.
      eiface
               \ library
               \ tests

Updated comments and added EIS references.

About FACEBOOK_APP_SPECIFIC_DATA_TEMPLATE, at the moment I decided to use a deferred class and then let the user redefine it, but maybe we can think in a different strategy for example provide a factory to load the specific data from an external source (a database, a json file, a property file).

### CODE REVIEW

File: library/facebook_app_specific_data.e [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+		end
+
+	api_secret :STRING
+			-- Facebook apps secret key. A ling string of characters
+		deferred
+		end
+
+	callback : STRING
+			-- For example yuor Facebook apps canvas adess. Read more at
+			-- https://developers.facebook.com/docs/facebook-login/manually-build-a-login-flow/
```
Comment (jocelyn) [inline library/facebook_app_specific_data.e:27]: minor comment, this could be using EIS info style

File: library/facebook_id_name.e [hunk 2]
```diff
@@ -19,12 +19,17 @@ feature {NONE} -- Creation
 		do
 			set_id (an_id)
 			create name.make_empty
+		ensure
+			id_set: id = an_id
+			id_set_to_zero: id.same_string ("0")
```
Comment (andersoxie) [inline library/facebook_id_name.e]: I do not understand the id_set_to_zero?

File: library/facebook_app_specific_data.e (hunk 1 above)
Comment (andersoxie) [inline library/facebook_app_specific_data.e:27]: Can you update the README file also decribing how the user should do with inheritance of the deferred class and the update in code necessary when they have created thier own application specific class?

Comment (jvelilla) [inline library/facebook_app_specific_data.e:27]: Yes, I have updated it, it should be part of the pull request now.

File: library/facebook_id_name.e (hunk 2 above)
Comment (jvelilla) [inline library/facebook_id_name.e]: Yes, it's wrong, thanks for pointing it.

File: library/facebook_app_specific_data.e (hunk 1 above)
Comment (jvelilla) [inline library/facebook_app_specific_data.e:27]: Hi, Anders,
It's already done.

Javier

File: README.md [hunk 3]
```diff
@@ -4,10 +4,9 @@ eiface
...
-
-Start with renaming the class FACEBOOK_APP_SPECIFIC_DATA_TEMÅLATE
- class to FACEBOOK_APP_SPECIFIC_DATA.
-Then add you Facebook app data in that class.
+Dependencies
+============
+You also need the the cypress library.
 
+First of all, you will need to create a new class.  It will need to inherit from
+FACEBOOK_APP_SPECIFIC_DATA and fill it with the specific data needed for Facebook API.
```
Comment (andersoxie) [inline README.md:12]: Do one need to call it a specific name or do one need to update the createion statement?

Comment (jvelilla) [inline README.md:12]: Here is an example of redefinition

https://github.com/jvelilla/eiface/blob/jv_review/tests/test_facebook_app_specific_data.e
and when you want to use it you can write.

create {TEST_FACEBOOK_APP_SPECIFIC_DATA} app_data

You can see the complete example here
https://github.com/jvelilla/eiface/blob/jv_review/tests/facebook_test.e

/Javier

**Pull request merged**